### ISSUE TITLE
Helm Resource Check fails due to (unclear) structure

### ISSUE THREAD

**Issue by tom-rm-meyer-ISST** [eclipse-tractusx/tractusx-quality-checks#49]
Not sure, if this issue is placed correctly because 
- this could also just be a TRG not being specific enough or 
- a workflow problem or 
- both

# Problem
In [puris/pr#33](https://github.com/eclipse-tractusx/puris/pull/33), we adopt the repo structure according to [TRG 2.03](https://eclipse-tractusx.github.io/docs/release/trg-2/trg-2-3) so that the root folder contains the charts. According to [TRG 5.02](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-02) the following structure is ok.

 ```
charts/
  puris-backend/
    .helmignore # mandatory  
	Chart.yaml # mandatory  
	LICENSE # mandatory  
	README.md # mandatory  
	values.yaml # mandatory  
	crds/ # optional  
	templates/ # mandatory, except of umbrella Helm charts  
	templates/NOTES.txt # optional
  puris-frontend/
    Chart.yaml
    ...
AUTHORS.md
DEPENDENCIES.md
LICENSE
README.md
```

In [this run](https://github.com/eclipse-tractusx/puris/actions/runs/6156697542/job/16705906207?pr=33) for the pr, the [TRG 5.04](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-04/) check fails BUT there are definitions for the resource management.

From what I can read in the workflow and the log, the chart is not found at charts/puris-backend, but there is charts/puris-backend/Chart.yaml

# Question
Is our structure wrong? We're not yet implementing the product helm, but created an issue for that. From my understanding (based on [helm doc](https://helm.sh/docs/chart_template_guide/getting_started/)) the target would be when using the dependency mechanism:
```
charts/
  puris
      charts/
          puris-backend
              Chart.yaml
                ..
          puris-.frontend
              Chart.yaml
               ...
          Chart.yaml
          ...
```

Is the intermediate structure (see above) wrong?

**Comment by tomaszbarwicki** [comment]
Hey @tom-rm-meyer-ISST, sorry for picking up your issue with a delay. 
Indeed, the puris-backend chart can't be read by quality check. There is a reason for that, unexpected character

https://github.com/eclipse-tractusx/puris/blob/7625dcdbe09c7e03d57a6a133b79fe447c3f4921/charts/puris-backend/values.yaml#L31

also by default, ingress should be disabled as you have it at frontend chart:

https://github.com/eclipse-tractusx/puris/blob/7625dcdbe09c7e03d57a6a133b79fe447c3f4921/charts/puris-backend/values.yaml#L92

I suggest to always check if 'helm template' is able to render chart. In this case it wasn't able to do that for backend in the beginning:

``` 
tractusx/puris/charts/puris-backend (main) $ helm template .
Error: cannot load values.yaml: error converting YAML to JSON: yaml: line 30: did not find expected key
```

**Comment by tom-rm-meyer-ISST** [comment]
Thanks a lot! I'll check that soon and come back to you in case of any further issues OR to close the PR :)

**Comment by Siegfriedk** [comment]
@tom-rm-meyer-ISST any update or can we close it?

**Comment by tom-rm-meyer-ISST** [comment]
@Siegfriedk I would like to keep it open until we fixed it for us with [PR#44](https://github.com/eclipse-tractusx/puris/pull/44#pullrequestreview-1679216821) to be 100% sure.

**Comment by tom-rm-meyer-ISST** [comment]
Thanks for reaching out and supporting us. We managed to fix it. Thanks a lot. Structure check is correctly working!
